### DISCUSSION TITLE
question about integrating snRNA seq data from different batches and different experimental conditions

### DISCUSSION THREAD

**Discussion by DLPFC** [satijalab/seurat#4899] [Q&A] [answered]
Hi all,

I have snRNA seq dataset obtained from 2 experimental conditions (stimulation vs control). For each condition, I have 3 different batches. Is there any way I can use integrate function to correct for batch effect as well as experimental condition at the same time? Seurat tutorial for integrate function just shows correcting for different experimental conditions.

Thanks

**Comment by timoast** [comment, marked as answer]
You can use the same functionality to integrate across batches, experimental conditions, or both.

**Reply by DLPFC** [reply]
Thanks for your reply.

Could you clarify if below steps make sense?
1) Merge all data
2) Split data by batch
3) Integrate
4) Take the integrated data and split by experimental condition
5) Integrate again

Thanks for your help!

**Reply by timoast** [reply]
No this doesn't make sense.

1. Split datasets by batch and condition
2. Integrate

**Reply by DLPFC** [reply]
Thank you. 

Since split function only takes a single attribute, would it make sense to create one attribute that contains both batch and condition information?

Thanks

**Reply by timoast** [reply]
Yes, if you have everything in a single object to begin with you can combine the batch and condition metadata information to create a new variable to split the object by.

Note that you just need to create a list of objects to integrate, and you don't need to use the `SplitObject()` function to do this. If you already have separate objects that need to be integrated, you can create a list simply using the `list()` function in R.

**Reply by DLPFC** [reply]
Thanks for your help!!
